src/Predictors/EventDrivenPredictor.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2016-2018 United States Government as represented by the
+// Administrator of the National Aeronautics and Space Administration.
+// All Rights Reserved.
+
+#include "Predictors/EventDrivenPredictor.h"
+#include "Contracts.h"
+#include "Messages/ProgEventMessage.h"
+
+namespace PCOE {
+    EventDrivenPredictor::EventDrivenPredictor(MessageBus& messageBus,
+                                               std::unique_ptr<Predictor>&& predictor,
+                                               std::string source)
+        : bus(messageBus), pred(std::move(predictor)), source(std::move(source)) {
+        Expect(pred, "Predictor pointer is empty");
+        bus.subscribe(this, this->source);
+    }
+
+    EventDrivenPredictor::~EventDrivenPredictor() {
+        bus.unsubscribe(this);
+    }
+
+    void EventDrivenPredictor::processMessage(const std::shared_ptr<Message>& message) {
+        switch (message->getMessageId()) {
+        case MessageId::ModelStateEstimate: {
```

Comment:
Can't we subscribe to ModelStateEstimate directly on line 15?

Reply:
I guess I was originally thinking that predictors might be interested in other messages, but that would require further changes to the predictor interface, so I think you're right.